### ISSUE TITLE
AWS account filter service-limits runs into unrefreshable check ID c1dfprch07

### ISSUE THREAD

**Issue by kjsch** [cloud-custodian/cloud-custodian#9001]
### Describe the bug

When running a Cloud Custodian policy using the service-limits filter for the account resource, the following error is encountered:

botocore.exceptions.ClientError: An error occurred (InvalidParameterValueException) when calling the RefreshTrustedAdvisorCheck operation: [UNREFRESHABLE_CHECK_ID_ERROR] Check c1dfprch07 is not refreshable.

### What did you expect to happen?

Expect the code to catch the exception when calling [refresh_trusted_advisor_check](https://github.com/cloud-custodian/cloud-custodian/blob/07c86ff215d9bc7a06687e9a3528ec659b0e5b80/c7n/resources/account.py#L822) and indicate in output that the check was not available/refreshable.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
$ custodian version --debug

Please copy/paste the following info along with any bug reports:

Custodian:   0.9.29
Python:      3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Platform:   win32
Using venv:  True
Docker: False
Installed:

argcomplete==3.1.1
attrs==23.1.0
boto3==1.28.5
botocore==1.31.5
docutils==0.18.1
importlib-metadata==5.2.0
jmespath==1.0.1
jsonschema==4.17.3
pyrsistent==0.19.3
python-dateutil==2.8.2
pyyaml==6.0.1
s3transfer==0.6.1
six==1.16.0
tabulate==0.9.0
urllib3==1.26.16
zipp==3.15.0
```

### Policy

```shell
policies:
  - name: service-limits
    resource: account
    filters:
      - type: service-limit
        names:
          - Lambda Code Storage Usage
        threshold: 85
```

### Relevant log/traceback output

```shell
Traceback (most recent call last):
  File "custodian.29\lib\site-packages\c7n\policy.py", line 330, in run
    resources = self.policy.resource_manager.resources()
  File "...\custodian.29\lib\site-packages\c7n\resources\account.py", line 79, in resources
    return self.filter_resources([get_account(self.session_factory, self.config)])
  File "...\custodian.29\lib\site-packages\c7n\manager.py", line 112, in filter_resources
    resources = f.process(resources, event)
  File "...\custodian.29\lib\site-packages\c7n\resources\account.py", line 868, in process
    matched = self.process_check(client, check, resources, event)
  File "...\custodian.29\lib\site-packages\c7n\resources\account.py", line 881, in process_check
    results = self.get_check_result(client, check['id'])
  File "...\custodian.29\lib\site-packages\c7n\resources\account.py", line 822, in get_check_result
    client.refresh_trusted_advisor_check(checkId=check_id)
  File "...\custodian.29\lib\site-packages\botocore\client.py", line 534, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "...\custodian.29\lib\site-packages\botocore\client.py", line 976, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidParameterValueException) when calling the RefreshTrustedAdvisorCheck operation: [UNREFRESHABLE_CHECK_ID_ERROR] Check c1dfprch07 is not refreshable.
```

### Extra information or context

We typically do not list specific names in the policy but let custodian query all of them.  This just started happening a couple days ago and only for this particular check ID.

**Comment by kjsch** [comment]
Actually, this error may have been happening for much longer, we just noticed it a couple days ago.  ;)

**Comment by kapilt** [comment]
do we know what that check id corresponds to in terms of actual check name?

**Comment by kapilt** [comment]
also noting we have a separate service quotas resource that's a bit more modern in terms of enabling monitoring and filing limit increases, the trusted advisor/ support case workflow in custodian pre-dates the existence of the service quotas api.

**Comment by kjsch** [comment]
The check name is 'Lambda Code Storage Usage'

I will look into service-quota, thanks

**Comment by palmobar** [comment]
I believe there are checks in trusted advisor that might not be available in service quotas (e.g. there are no related metrics). Please correct me if I'm wrong here.

**Comment by ajkerrigan** [comment]
We got another report of this [in Slack](https://www.linen.dev/s/cloud-custodian/t/15983568/hey-guys-i-have-run-into-an-issue-that-is-documented-on-the-#87a83184-6932-48f8-a2ce-c0b137c2550c).

It does sound like policies should prefer service-quota where possible, but like @palmobar said that doesn't cover everything. If we warn/continue on non-refreshable checks, would it be worth pointing out the service-quota resource type in the log message?

**Comment by ajkerrigan** [comment]
This should be fixed via #9072 - we can reopen if there are any lingering issues.
